R/utils.R

```diff
@@ -117,6 +117,37 @@ do_patterns = function(pat_sub, all_cols) {
   return(matched)
 }
 
+
+generate_yaml = function(x, col.names = TRUE, sep = ",", sep2 = "", 
+                          eol = if (.Platform$OS.type=="windows") "\r\n" else "\n",
+                          na = "", dec = "", qmethod = "double", 
+                          logical01 = getOption("datatable.logical01", FALSE)) {
+  schema_vec = sapply(x, class)
+  # multi-class objects reduced to first class
+  if (is.list(schema_vec)) schema_vec = sapply(schema_vec, `[`, 1L)
+  # as.vector strips names
+  schema_vec = list(name=names(schema_vec), type=as.vector(schema_vec))
```

Comment:
I'm looking at the code for `schema_vec` again and it's a bit of a mess.

What about `schema_vec = sapply(x, class, simplify = FALSE)`

Then in the `seq_along` loop we use the names of this object and the first values:

```
list(
  name = names(schema_vec)[i],
  type = schema_vec[[ii]][1L]
)
```

Reply:
I hadn't really look at it (just copy-pasted it into the new function). However, I see now that we are reducing multiclass objects to their first class. Why is that? Can't we store class as a character vector? It is supported by yaml, isn't it?

Reply:
I was thinking the same thing on my first re-read. I'm not sure if `fread` would support that though since `colClasses` needs length-1 classes;

OTOH maybe it's best to make `fread` to the `[1]` and leave the full class in there from `fwrite`

Reply:
Maybe it's a discussion for another PR. For the time being, I left it as it is.